src/validation.cpp

```diff
@@ -2913,18 +2927,23 @@ bool CChainState::ConnectBlock(const CBlock& block, CValidationState& state, CBl
             minerAddress = GetScriptForDestination(dest);
         }
 
-        const auto blockResult = evm_finalize(evmContext, true, block.nBits, beneficiary, block.GetBlockTime());
+        const auto evmBlockResult = evm_finalize(evmContext, true, block.nBits, beneficiary, block.GetBlockTime());
+        auto evmBlockHash = std::vector<uint8_t>(evmBlockResult.block_hash.begin(), evmBlockResult.block_hash.end());
+        std::reverse(evmBlockHash.begin(), evmBlockHash.end());
+
+        mnview.SetBlockHash(CEvmDvmMapType::DvmEvm, block.GetHash(), uint256(evmBlockHash));
```

Comment:
Resolve conflicts and move to masternodes/validation.cpp calling SetBlockHash on cache, on which changes are added to the block undo.